src/java/fr/paris/lutece/plugins/searchstats/business/QueryRecordDAO.java

```diff
@@ -186,4 +196,28 @@ private String buildQuery( String strSelect, List<String> clauses )
 
         return strSQL;
     }
+
+    public List<QueryRecord> selectQueryRecordList( Plugin plugin )

```

Comment:
How big this list can be ? It seems that you never use the full list anyway. To limit risks of blowing up memory, I think you should pass down restrictions to the search down to here (top n and date criteria).
Maybe extend `RecordFilter` with your needs, and use `selectQueryRecordListByCriteria`